feature-libs/cart/main/components/cart/clear-cart/clear-cart.service.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { TestBed } from '@angular/core/testing';
+import { ActiveCartFacade } from '@spartacus/cart/main/root';
+import { Observable, of } from 'rxjs';
+import { takeLast, take } from 'rxjs/operators';
+import { ClearCartService } from './clear-cart.service';
+import { GlobalMessageService, GlobalMessageType } from '@spartacus/core';
+import createSpy = jasmine.createSpy;
+
+class MockGlobalMessageService implements Partial<GlobalMessageService> {
+  add = createSpy().and.stub();
+}
+
+class MockActiveCartFacade implements Partial<ActiveCartFacade> {
+  addEntry(_productCode: string, _quantity: number): void {}
+  clearActiveCart(): Observable<boolean> {
+    return of();
+  }
+}
+
+describe('ClearCartService', () => {
+  let service: ClearCartService;
+  let activeCartService: ActiveCartFacade;
+  let globalMessageService: GlobalMessageService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        { useClass: MockActiveCartFacade, provide: ActiveCartFacade },
```

Comment:
That's just me nitpicking, for readability, as a convention we start these with `provide:`first.